File: dart/dojo/kata8.dart
// https://www.codewars.com/kata/58f5c63f1e26ecda7e000029/train/dart

List<String> wave(String str) {
  List<String> solution = [];
  for (var i = 0; i < str.length; i++) {
    var a = str.substring(0, i);
    var b = new String.fromCharCode(str.runes.elementAt(i)).toUpperCase();
    var c = str.substring(i + 1);
    solution.add(a + b + c);
  }
  while (solution.remove(str)) {}
  print(solution);
  return solution;
}

// List<String> anotherSolution(String str) {
//   return List.filled(str.length, str)
//       .asMap()
//       .entries
//       .where((e) => e.value[e.key] != " ")
//       .map((e) =>
//           e.value.replaceRange(e.key, e.key + 1, e.value[e.key].toUpperCase()))
//       .toList();
// }

main(List<String> args) {
  wave("hello");
  // wave(" gab ");
}

File: kotlin/dojo/kata8.kt
// https://www.codewars.com/kata/56541980fa08ab47a0000040/train/kotlin

fun printerError(s: String): String {
    val a = "abcdefghijklm"
    var e = 0
    s.forEach {
        println(it)
        if(it !in a) e++
    }
    var solution = "$e/${s.length}"
    println(solution)
    return solution
}  

fun main(args: Array<String>) {
    val e = "aaaxbbbbyyhwawiwjjjwwm"
    val a = "8/22"
    printerError(e)
}

// fun printerError(s: String) = "${s.count { it !in 'a'..'m' }}/${s.length}"
// fun printerError(s: String): String {
//     return "${s.count{it > 'm'}}/${s.length}"
// }
